lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -208,7 +220,21 @@ def add_included_resources_for(serializer, include_tree, primary_data, included)
           end
         end
 
-        def links_for(serializer, options)
+        def links_for(serializer)
+          serializer.links.each_with_object({}) do |(name, value), hash|
+            hash[name] =
+              if value.respond_to?(:call)
```

Comment:
Basic duck-typing for a callable object (mainly `Proc` or `Lambda`). Anyways the value should either be a callable or a `String`.